plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/AlertsWidget.java

```diff
@@ -0,0 +1,159 @@
+package edu.wpi.first.shuffleboard.plugin.base.widget;
+
+import edu.wpi.first.shuffleboard.api.widget.Description;
+import edu.wpi.first.shuffleboard.api.widget.ParametrizedController;
+import edu.wpi.first.shuffleboard.api.widget.SimpleAnnotatedWidget;
+import edu.wpi.first.shuffleboard.plugin.base.data.AlertsData;
+import edu.wpi.first.shuffleboard.plugin.base.data.AlertsData.AlertItem;
+import edu.wpi.first.shuffleboard.plugin.base.data.AlertsData.AlertType;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.control.MultipleSelectionModel;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.Pane;
+import javafx.scene.text.TextAlignment;
+
+@Description(name = "Alerts", dataTypes = AlertsData.class, summary = "Displays a list of alerts.")
+@ParametrizedController("AlertsWidget.fxml")
+public final class AlertsWidget extends SimpleAnnotatedWidget<AlertsData> {
+
+  private Image errorIcon = new Image(getClass().getResourceAsStream("icons/error.png"));
+  private Image warningIcon = new Image(getClass().getResourceAsStream("icons/warning.png"));
+  private Image infoIcon = new Image(getClass().getResourceAsStream("icons/info.png"));
+
+  @FXML
+  private Pane root;
+
+  @FXML
+  private ListView<AlertItem> list;
+
+  @FXML
+  @SuppressWarnings("incomplete-switch")
+  private void initialize() {
+    list.setCellFactory(param -> new ListCell<AlertItem>() {
+      @Override
+      protected void updateItem(AlertItem item, boolean empty) {
+        super.updateItem(item, empty);
+        if (empty || item == null) {
+          setGraphic(null);
+          setText(null);
+        } else {
+          setMinWidth(param.getWidth() - 32);
+          setMaxWidth(param.getWidth() - 32);
+          setPrefWidth(param.getWidth() - 32);
+
+          setWrapText(true);
+          setText(item.text);
+
+          if (item.type == AlertType.LOCAL) {
+            setTextAlignment(TextAlignment.CENTER);
+            setStyle("-fx-alignment: center;");
+            setGraphic(null);
+          } else {
+            setTextAlignment(TextAlignment.LEFT);
+            setStyle("-fx-alignment: left;");
+
+            ImageView imageView = new ImageView();
+            switch (item.type) {
+              case ERROR:
+                imageView.setImage(errorIcon);
+                break;
+              case WARNING:
+                imageView.setImage(warningIcon);
+                break;
+              case INFO:
+                imageView.setImage(infoIcon);
+                break;
+              default:
+                break;
+            }
```

Comment:
Java 17 language features make this a lot nicer

```java
var icon = switch (item.type) {
  case ERROR -> errorIcon;
  case WARNING -> warningIcon;
  case INFO -> infoIcon;
}

imageView.setImage(icon);
```

Reply:
It looks like we would need to update to at least checkstyle 8.36 to support Java 14 switches, but there are compatibility issues with the current config files. It seems like that should be fixed separately, but is probably not blocking?

Reply:
Yeah, that could be fixed separately